sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala

```diff
@@ -260,6 +260,65 @@ class HiveCatalogedDDLSuite extends DDLSuite with TestHiveSingleton with BeforeA
       assert(err.contains("Cannot recognize hive type string:"))
    }
   }
+
+  test("SPARK-26630: Fix ClassCastException in TableReader while creating HadoopRDD") {
+    withTable("table_old", "table_pt_old", "table_new", "table_pt_new",
+      "table_ctas_old", "table_ctas_pt_old", "table_ctas_new", "table_ctas_pt_new") {
```

Comment:
Let us split this test case to multiple smaller ones.

Reply:
Done, split it into 4 test cases.